### ISSUE TITLE
Rework Error 3 popup

### ISSUE THREAD

**Issue by fynngodau** [corona-warn-app/cwa-wishlist#464]
### Current Implementation

On devices without Google Play services, the Error 3 popup keeps popping up, annoying users.

![error3](https://user-images.githubusercontent.com/16943720/115239645-42b2bc00-a11f-11eb-9447-277928be42da.png)

### Suggested Enhancement

In the order of work required (descending), and thereby also in the order of improvement of the app (ascending):

1. Reword the popup. It is false that users without Google Play services "cannot use the Corona-Warn-App", specifically considering the possibility of checking in.
2. Pop it up less. Nobody likes using an app that keeps popping up dialogs.
3. Improve the flow. Integrate the warning with the information into onboarding, instead of allowing users to tap Enable Exposure Logging, then show a warning card on the home screen informing users of the status of the app.

### Expected Benefits

1. Users who are not using Google Play services can use Corona-Warn-App to check in at events and locations, to query statistics regarding the current infection rates in Germany, to use the Contact Journal and to retrieve their test results.
2. Developers who are trying out the app in an emulator without Google Play services are not annoyed.

**Comment by MikeMcC399** [comment]
@fynngodau 
I would enjoy seeing this implemented, especially for the emulator environment.

Perhaps it would be better placed in the Android repository, since it is not relevant for iOS?

**Comment by fynngodau** [comment]
Since it's a rather big suggestion with multiple different ideas, I imagined it would be a better fit for the wishlist repository. However if this is in fact better fit for the Android repository, somebody can of course feel free to move it there.

**Comment by jucktnich** [comment]
I think, you can't use the app on iOS 13-13.7 due to the lack of ENF v 2. So a discussion about this topic does also make sense for iOS, so the wishlist is the right place.

**Comment by Ein-Tim** [comment]
The main issue for the availability is: #443

**Comment by MikeMcC399** [comment]
@jucktnich / @Ein-Tim 

What is the error message on iOS if ENF is not available? Does the app keep popping up an error message like on Android?
https://www.coronawarn.app/en/faq/#17 is specific to Google/Android.

**Comment by jucktnich** [comment]
If the iOS version has no  ENF v 2 there's an update os screen, which you can't close I think.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

The app shows:

![](https://user-images.githubusercontent.com/10122052/105043882-60ac4a00-5a66-11eb-846b-0c02566ddc37.png)

**Comment by MikeMcC399** [comment]
Ein-Tim 
Thank you very much for the iOS screen shot! That makes it clear. It seems like the problems on Android and iOS are quite different in that case.

**Comment by Ein-Tim** [comment]
Yes, I would also suggest to not mix the Android Error 3 pop up together with the availability on older devices, which is tracked in #443

**Comment by jucktnich** [comment]
I think the reason for the two messages are pretty much the same, so I would suggest to find a a same solution for both.

**Comment by heinezen** [comment]
@fynngodau and everyone else,

I have taken all your feedback and mirrored it to Jira as ticket [EXPOSUREAPP-6594](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6594). The development team and the RKI will take a look at this.

----
Corona-Warn-App Open Source Team

**Comment by vaubaehn** [comment]
@fynngodau and all - concerning Android:
Isn't the pop up dialog triggered with every API call to ENS (causing an exception)?
Imho, this is not so different from https://github.com/corona-warn-app/cwa-app-android/pull/2873 , the Bluetooth-based Incompatibility. Couldn't the exception be persisted and a Google Play Services Incompatibility card be shown? I think even the texts would only need minor adjustments.
What do you think?

**Comment by dsarkar** [comment]
@fynngodau 
Actually you could another point in your OP:
user might wanna use the check-in feature.

**Comment by Ein-Tim** [comment]
@dsarkar 

This is tracked in #443

**Comment by fynngodau** [comment]
@dsarkar This is already included under Expected benefits and in fact one of the main points why I opened this issue:

> Users who are not using Google Play services can use Corona-Warn-App to check in at events and locations

(Though I forgot to update this to also apply to iOS)

**Comment by dsarkar** [comment]
@fynngodau You are right! Apologies, must have skipped it!

**Comment by vaubaehn** [comment]
@fynngodau 
From your OP:
> ### Suggested Enhancement
> [...]
> 3. Improve the flow. Integrate the warning with the information into onboarding, instead of allowing users to tap Enable Exposure Logging, **then show a warning card on the home screen** informing users of the status of the app.

That's now also in the mind of the devs: https://github.com/corona-warn-app/cwa-app-android/pull/2995#discussion_r622277201
❤️ 

Let's hope there will be a more relaxed situation later, that there is enough time for them to go on on that issue.

**Comment by MikeMcC399** [comment]
@vaubaehn 

Thanks for the pointer to PR https://github.com/corona-warn-app/cwa-app-android/pull/2995.

- I was able to run the pre-release 2.1 app without getting any errors on a device with ENS available, provided I did not try to activate exposure logging. 
- On a Pixel 3a emulator with no Google Play services installed, I get a warning 
"Update Google Play services
CWA TEST won't run unless you update Google Play services." If I tap on the notification, the app opens and the functions, apart from exposure logging seem to work. If I do not tap on the notification, the app just appears with a header and nothing below on the screen.

That is a good step forward. It could be made more user-friendly though if the app is going to be supported in future in an environment without Google Play services.

**Comment by fynngodau** [comment]
@vaubaehn Thanks for linking the PR, it sounds like it implements my suggestion number 2 for Android ("Pop up Android popup less").
